core/src/main/java/io/substrait/extension/ImmutableExtensionLookup.java

```diff
@@ -30,14 +31,16 @@ public static class Builder {
     private final Map<Integer, SimpleExtension.FunctionAnchor> functionMap = new HashMap<>();
     private final Map<Integer, SimpleExtension.TypeAnchor> typeMap = new HashMap<>();
 
-    public Builder from(Plan p) {
+    public Builder from(
```

Comment:
Nice refactor! we could also overload the API and add `from(Plan p)` and `from(ExtendedExpression e)` that call `from(simpleExtensionURIs, simpleExtensionDeclarations)`.

Reply:
+1 to keeping the existing `from(Plan p)` and having it call your newly refactor method.

Reply:
Added